### ISSUE TITLE
pouchDB with Angular2

### ISSUE THREAD

**Issue by sebastian-leidig** [NGO-DB/ndb-core#9]
Previously we used [angular-pouchdb](https://github.com/angular-pouchdb/angular-pouchdb) (and [pouchdb-authentication plugin](https://github.com/nolanlawson/pouchdb-authentication)) to access and locally duplicate the database. `angular-pouchdb` is not available for Angular 2 currently, (also see https://github.com/angular-pouchdb/angular-pouchdb/issues/94) so we some workaround.

Let's look into `ngUpgrade` how to wrap the Angular 1 module into an Angular 2 service:
- https://angular.io/docs/ts/latest/guide/upgrade.html#!#making-angular-1-dependencies-injectable-to-angular-2
- http://blog.thoughtram.io/angular/2015/10/24/upgrading-apps-to-angular-2-using-ngupgrade.html

**Comment by sebastian-leidig** [comment]
TODO: Should we load PouchDB libraries dynamically rather than including them in index.html?

**Comment by sebastian-leidig** [comment]
Solved using DefinitelyTyped for PouchDB and loading the library in index.html 
PouchDB is working smoothly now.

We encapsulated Authentication and simple CRUD in two classes `DatabaseManagerService` and `Database`. The concrete implementations using PouchDB are hidden behind abstract classes.

![architecture_core](https://cloud.githubusercontent.com/assets/1682541/15628012/acd6a7aa-24f5-11e6-85b8-959b03cb5191.png)
